src/composeappmanager.cc

```diff
@@ -193,7 +194,8 @@ ComposeAppManager::AppsContainer ComposeAppManager::getApps(const Uptane::Target
   return apps;
 }
 
-ComposeAppManager::AppsContainer ComposeAppManager::getAppsToUpdate(const Uptane::Target& t) const {
+ComposeAppManager::AppsContainer ComposeAppManager::getAppsToUpdate(
+    const Uptane::Target& t, std::unordered_map<std::string, std::string>& apps_and_reasons) const {
```

Comment:
Maybe using `AppsContainer` type for `apps_and_reasons` instead of `std::unordered_map<std::string, std::string>` would make the change/code a bit cleaner?

Reply:
I though about it. Problem is that currently `AppsContainer` contains `name` + `uri`. We could re-structure `AppsContainer` to make it contain an additional field, but it would be unset most of the time, and additional code changes would be needed. Just replacing the use of `uri` with `reason` in that specific case would be too confusing, IMO.

Reply:
> We could re-structure `AppsContainer` to make it contain an additional field, but it would be unset most of the time

Why do we need any additional field in `AppsContainer`?

Effectively, `AppsContainer == std::unordered_map<std::string, std::string>`, this is exactly the type you use for the second param `apps_and_reasons`.
Therefore, `getAppsToUpdate(
    const Uptane::Target& t, std::unordered_map<std::string, std::string>& apps_and_reasons)` can be declared as `getAppsToUpdate(
    const Uptane::Target& t, AppsContainer& apps_and_reasons)`

Reply:
I think reusing the `AppsContainer` type replacing the `uri` with `reason` as the `.second` field might led to confusion. I would expect `AppsContainer` to be used consistently in all locations.

Reply:
Well, I don't see much problem in it as long as variable is named correspondingly. We use the same type, e.g. `std::unordered_map<std::string, std::string>` for many different things, and what exactly the container carries is determined rather by the usage context and variable name.

As an option, we can introduce a new type, e.g. `AppsSyncReason == std::unordered_map<std::string, std::string>` or something like that.

Reply:
I'm OK with having  `AppsSyncReason`.
I've updated the commit accordingly.

Reply:
Thanks.